specs/deneb/p2p-interface.md

```diff
@@ -138,46 +151,62 @@ Per `context = compute_fork_digest(fork_version, genesis_validators_root)`:
 | `ALTAIR_FORK_VERSION`    | `altair.SignedBeaconBlock`    |
 | `BELLATRIX_FORK_VERSION` | `bellatrix.SignedBeaconBlock` |
 | `CAPELLA_FORK_VERSION`   | `capella.SignedBeaconBlock`   |
+| `DENEB_FORK_VERSION`     | `deneb.SignedBeaconBlock`     |
+
+No more than `MAX_REQUEST_BLOCKS_DENEB` may be requested at a time.
+
+#### BlobSidecarsByRoot v1
 
-#### BeaconBlockAndBlobsSidecarByRoot v1
+**Protocol ID:** `/eth2/beacon_chain/req/blob_sidecars_by_root/1/`
 
-**Protocol ID:** `/eth2/beacon_chain/req/beacon_block_and_blobs_sidecar_by_root/1/`
+New in deneb.
 
 Request Content:
 
+```python
+class BlobIdentifier(Container):
+    block_root: Root
+    index: uint64
+```
+
 ```
 (
-  List[Root, MAX_REQUEST_BLOCKS]
+  List[BlobIdentifier, MAX_REQUEST_BLOBS_SIDECARS * MAX_BLOBS_PER_BLOCK]
```

Comment:
Why do the multiplier here? Just so we can get a 1:1 symmetry between blocks requests and blobs request (the same number of blocks requests could get the same number of dependent blobs)

Do you suspect that the usecases of blobsbyroot would need to actually use a >128 capacity in the normal case or do they normally make much smaller requests to fill in smaller pieces of missing info?

Reply:
> Just so we can get a 1:1 symmetry between blocks requests and blobs request

Yes: the resources we're spending is network bandwidth and memory so we might as well keep the analysis simple, as far as upper-limits go. The choice is somewhat arbitrary, as long as there exists _a_ reasonable limit that clients are reasonably able to keep in memory while stream..

> Do you suspect that the usecases of blobsbyroot would need to actually use a >128 capacity in the normal case or do they normally make much smaller requests to fill in smaller pieces of missing info?

Not really, but it also doesn't hurt - the key point we should strive for is to keep byroot block limit in sync with byroot blob so that block+blob requests (for example when seeing an attestation for an unknown block) can use the same paging worry-free.

It also wouldn't hurt to have a (much) lower limit for byroot in general (ie 4 or 8 blocks+corresponding blobs) but per above point, we don't gain much in terms of total security due to byrange - no point wasting precious spec lines on a separate constant, imo.